ui/redux/actions/content.js

```diff
@@ -151,47 +209,187 @@ export function doDownloadUri(uri: string) {
+      } else {
+        if (collection.collectionId === playingCollection.collectionId) {
+          // keep current playingCollection data like loop or shuffle if plpaying the same
+          dispatch(
+            doSetPlayingUri({ ...playingUri, ...playingOptions, collection: { ...playingCollection, ...collection } })
+          );
+        } else {
+          dispatch(
+            doSetPlayingUri({ ...playingUri, ...playingOptions, collection: willPlayCollection ? collection : {} })
+          );
+        }
+      }
+    }
 
-    if (!isFloating && !source) dispatch(doSetPrimaryUri(uri));
+    if (!isLive) {
+      const isAuthenticated = selectUserVerifiedEmail(state);
+      const playCb = isAuthenticated ? (fileInfo) => dispatch(doAnaltyicsPurchaseEvent(fileInfo)) : undefined;
+      dispatch(doPlayUri(uri, false, true, playCb, willPlayCollection));
+    }
 
-    if (isPlayable) dispatch(doSetPlayingUri(playingOptions));
+    if (cb) cb(uri);
+  };
+}
 
-    if (!isLive) dispatch(doPlayUri(uri, false, true, (fileInfo) => dispatch(doAnaltyicsPurchaseEvent(fileInfo))));
+export function doPlaylistAddAndAllowPlaying({
+  uri,
+  collectionName,
+  collectionId: id,
+  createNew,
+  push,
+}: {
+  uri?: string,
+  collectionName: string,
+  collectionId?: string,
+  createNew?: boolean,
+  push?: (uri: string) => void,
+}) {
+  return (dispatch: Dispatch, getState: () => any) => {
+    const state = getState();
+    const remove = Boolean(id && uri && selectCollectionForIdHasClaimUrl(state, id, uri));
+
+    let collectionId = id;
+    if (createNew) {
+      dispatch(
+        doLocalCollectionCreate({ name: collectionName, items: uri ? [uri] : [], type: 'playlist' }, (newId) => {
+          collectionId = newId;
+        })
+      );
+    } else if (collectionId && uri) {
+      dispatch(doCollectionEdit(collectionId, { uris: [uri], remove, type: 'playlist' }));
+    }
+
+    if (!uri && createNew) return;
+
+    const collectionPlayingId = selectPlayingCollectionId(state);
+    const playingUri = selectPlayingUri(state);
+    const isUriPlaying = uri && selectIsUriCurrentlyPlaying(state, uri);
+    const firstItemUri = selectFirstItemUrlForCollection(state, collectionId);
+
+    const isPlayingCollection = collectionPlayingId && collectionId && collectionPlayingId === collectionId;
+    const hasItemPlaying = playingUri.uri && !isUriPlaying;
+    const floatingPlayerEnabled =
+      playingUri.collection.collectionId === 'queue' || selectClientSetting(state, SETTINGS.FLOATING_PLAYER);
+
+    const startPlaying = () => {
+      if (isUriPlaying) {
+        dispatch(doSetPlayingUri({ ...playingUri, collection: { collectionId } }));
+      } else {
+        dispatch(
+          doUriInitiatePlay(
+            { uri: createNew ? uri : firstItemUri, collection: { collectionId } },
+            true,
+            true,
+            !floatingPlayerEnabled && push ? (url) => push(url) : undefined
+          )
+        );
+      }
+    };
```

Comment:
- [ ] :exclamation: I added a video into a private collection, and clicked "Start playing" from the toast. It ended up playing Watch Later, which wasn't the target.
- [ ] :exclamation: Create a new private playlist, then add 1 item into from the homepage. Start Playing doesn't work until there are 2 items in the list.

Reply:
still happens?

Reply:
The second item is still happening